onnxruntime/core/providers/cpu/math/shrink.cc

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cpu/math/shrink.h"
+
+#include "core/util/math.h"
+#include "core/util/math_cpuonly.h"
+#include <iostream>
+
+namespace onnxruntime {
+ONNX_CPU_OPERATOR_KERNEL(
+    Shrink,
+    9,
+    KernelDefBuilder().TypeConstraint("T", {DataTypeImpl::GetTensorType<float>(),
```

Comment:
>DataTypeImpl::GetTensorType [](start = 44, length = 27)

Can you use DataTypeImpl::AllFixedSizeTensorTypes?

Reply:
Any reason not to use the aggregate type instead of maintaining such a long list?

Reply:
I was using #456 as my reference PR and hence followed the same template. Also, DataTypeImpl::AllFixedSizeTensorType() seems to contain `bool` which is not supported by this op. Going along the same lines, I have a new type aggregation DataTypeImpl::AllNumericTensorTypes() which should do the trick and remove the long list in this file.

Reply:
awesome. thanks!